### PR TITLE
Add `eslint-plugin-perfectionist` and `eslint-plugin-react-perf` to the base and `/react` configs, respectively

### PR DESCRIPTION
**Pull request by mangs**
**Pull Request Checklist**

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) document
- [x] Readme and changelog updates were made reflecting this PR's changes
- [x] Increase the project version number in `package.json` following [Semantic Versioning](http://semver.org/)

**Changes Included**

- Version bump to `2.1.0`
- Add [`eslint-plugin-perfectionist`](https://github.com/azat-io/eslint-plugin-perfectionist) to the base configuration and all its dependent configurations
  - Provides more robust and consistent alphabetical sorting features than what ESLint provides by default
  - All rules are auto-fixable
  - Its rules replace rules `import/order`, `sort-keys`, and `sort-vars`
- Add [`eslint-plugin-react-perf`](https://github.com/cvazac/eslint-plugin-react-perf) to the `/react` configuration and all its dependent configurations
  - Provides performance-related checks to prevent code that causes performance problems
- Upgrade dependencies to latest versions
- Upgrade Bun version from `1.1.17` to `1.1.27`

### CODE REVIEW

File: lib/eslintBunConfig.json [hunk 1]
```diff
@@ -3,12 +3,31 @@
...
-  },
   "rules": {
     "n/no-unsupported-features/es-builtins": "off",
     "n/no-unsupported-features/es-syntax": "off",
-    "n/no-unsupported-features/node-builtins": "off"
+    "n/no-unsupported-features/node-builtins": "off",
+    "perfectionist/sort-imports": [
+      "error",
+      {
+        "environment": "bun",
```
Comment (mangs) [inline lib/eslintBunConfig.json:13]: `environment` switched to Bun so `bun:*` imports work (e.g. `bun:test`) without any more hacks (e.g. the `settings:` section that I removed in this PR)

File: lib/eslintBaseConfig.json [hunk 2]
```diff
@@ -33,24 +35,34 @@
...
+            "builtin-type",
+            "external-type",
+            "index-type",
+            "internal-type",
+            "parent-type",
+            "sibling-type",
+            "type"
+          ],
+          "style"
+        ]
```
Comment (mangs) [inline lib/eslintBaseConfig.json:62]: a huge upside to these are they are now auto-fixable with `eslint-plugin-perfectionist`, so you can fix your whole codebase with one command (e.g. `eslint --fix`)

File: lib/eslintReactConfig.json [hunk 3]
```diff
@@ -1,9 +1,10 @@
 {
-  "plugins": ["eslint-plugin-jsx-a11y"],
+  "plugins": ["eslint-plugin-jsx-a11y", "eslint-plugin-react-perf"],
   "extends": [
     "eslint-config-airbnb",
     "airbnb/hooks",
     "plugin:jsx-a11y/recommended",
+    "plugin:react-perf/recommended",
```
Comment (mangs) [inline lib/eslintReactConfig.json:7]: nice performance recommendations here for React apps

File: test/eslintBaseConfig/exampleBaseConfigEntrypoint.mjs
Comment (mangs) [inline test/eslintBaseConfig/exampleBaseConfigEntrypoint.mjs:1]: fix tests from here down

**Review by mangs — COMMENTED** [review]
Some notes for context:

File: test/eslintBaseConfig/exampleBaseConfigEntrypoint.mjs
Comment (mangs) [inline test/eslintBaseConfig/exampleBaseConfigEntrypoint.mjs:1]: fix tests from here down

File: test/eslintBaseConfig/exampleBaseConfigEntrypoint.mjs [hunk 4]
```diff
@@ -192,26 +192,26 @@ export {
   anError,
```
Comment (mangs) [inline test/eslintBaseConfig/exampleBaseConfigEntrypoint.mjs:192]: text fixes from here down

**Review by jduthon — APPROVED** [review]

**Pull request merged**